src/deepsparse/loggers/metric_functions/cv/built_ins.py

```diff
@@ -0,0 +1,273 @@
+from typing import Dict, List, Optional, Tuple, Union
+
+import numpy
+
+
+__all__ = [
+    "image_shape",
+    "mean_pixels_per_channel",
+    "std_pixels_per_channel",
+    "mean_score_per_detection",
+    "std_score_per_detection",
+    "fraction_zeros",
+    "count_classes_detected",
+    "count_number_objects_detected",
+]
+
+
+def image_shape(
+    img: Union[numpy.ndarray, "torch.tensor"]  # noqa F821
+) -> Dict[str, int]:
+    """
+    Return the shape of the image.
+
+    :param img: An image represented as a numpy array or a torch tensor.
+        Assumptions:
+            - 3 dimensional or 4 dimensional (num_batches in zeroth dimension)
+              tensor/array
+            - the image has 3 or 1 channels
+    :return: Dictionary that maps "dim_0", "dim_1" and "channels" keys to the
+        appropriate integers
+    """
+    img_numpy = _assert_numpy_image(img)
+    num_dims, channel_dim = _check_valid_image(img_numpy)
+    if num_dims == 4:
+        img_numpy = img_numpy[0]
+        channel_dim -= 1
+
+    result = {"channels": img_numpy.shape[channel_dim]}
+    dims_counter = 0
+    for index, dim in enumerate(img_numpy.shape):
+        if index != channel_dim:
+            result[f"dim_{dims_counter}"] = dim
+            dims_counter += 1
+    return result
+
+
+def mean_pixels_per_channel(
+    img: Union[numpy.ndarray, "torch.tensor"]  # noqa F821
+) -> Dict[str, float]:
+    """
+    Return the mean pixel value per image channel
+
+    :param img: An image represented as a numpy array or a torch tensor.
+        Assumptions:
+            - 3 dimensional or 4 dimensional (num_batches in zeroth dimension)
+              tensor/array
+            - the image has 3 or 1 channels
+    :return: Dictionary that maps channel number to the mean pixel value
+    """
+    img_numpy = _assert_numpy_image(img)
+    num_dims, channel_dim = _check_valid_image(img_numpy)
+    dims = numpy.arange(0, num_dims, 1)
+    dims = numpy.delete(dims, channel_dim)
+    means = numpy.mean(img_numpy, axis=tuple(dims))
+    keys = ["mean_channel_{}".format(i) for i in range(len(means))]
+    return dict(zip(keys, means))
+
+
+def std_pixels_per_channel(
+    img: Union[numpy.ndarray, "torch.tensor"]  # noqa F821
+) -> Dict[str, float]:
+    """
+    Return the standard deviation of pixel values per image channel
+    :param img: An image represented as a numpy array or a torch tensor.
+        Assumptions:
+            - 3 dimensional or 4 dimensional (num_batches in zeroth dimension)
+              tensor/array
+            - the image has 3 or 1 channels
+    :return: Dictionary that maps channel number to the std pixel value
+    """
+    img_numpy = _assert_numpy_image(img)
+    num_dims, channel_dim = _check_valid_image(img)
+    dims = numpy.arange(0, num_dims, 1)
+    dims = numpy.delete(dims, channel_dim)
+    stds = tuple(numpy.std(img_numpy, axis=tuple(dims)))
+    keys = ["std_channel_{}".format(i) for i in range(len(stds))]
```

Comment:
do we need `std_` here? shouldn't we already be nested under the `std` function?

Reply:
as in e.g.`identifier.remainder__std_pixels_per_channel__channel_0`? Yes, you are right.